x-pack/plugins/maps/public/index_pattern_util.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IFieldType, IndexPattern } from 'src/plugins/data/public';
+import { i18n } from '@kbn/i18n';
+import { getIndexPatternService, getIsGoldPlus } from './kibana_services';
+import { indexPatterns } from '../../../../src/plugins/data/public';
+import { ES_GEO_FIELD_TYPE, ES_GEO_FIELD_TYPES } from '../common/constants';
+
+export function getGeoTileAggNotSupportedReason(field: IFieldType): string | null {
+  if (!field.aggregatable) {
+    return i18n.translate('xpack.maps.geoTileAgg.disabled.docValues', {
+      defaultMessage: 'Clustering requires doc_values.',
+    });
+  }
+
+  if (field.type === ES_GEO_FIELD_TYPE.GEO_SHAPE && !getIsGoldPlus()) {
+    return i18n.translate('xpack.maps.geoTileAgg.disabled.license', {
+      defaultMessage: 'Geo_shape clustering requires a Gold license.',
```

Comment:
maybe reword to `geo_shape clustereing requires a Gold license or higher`

Reply:
Talked to Gail, she did not like "or higher" phase. Alerting does not use this.

![image](https://user-images.githubusercontent.com/373691/84439824-2b996880-abf6-11ea-9c46-25abec5fdf9c.png)